sqe-http-api/Controllers/UserController.cs

```diff
@@ -20,34 +23,126 @@ public UserController(IUserService userServiceAuthenticate)
         {
             _userService = userServiceAuthenticate;
         }
+        
+        /// <summary>
+        /// Provides the user details for a user with valid JWT in the Authorize header
+        /// </summary>
+        [HttpGet]
+        public ActionResult<UserDTO> GetCurrentUser()
+        {
+            var user = _userService.GetCurrentUser();
+
+            if (user == null)
+                return Unauthorized(new { message = "No current user", code =601 }); // TODO: Add Error Code
+
+            return user;
+        }
+        
+        /// <summary>
+        /// Creates a new user with the submitted data.
+        /// </summary>
+        /// <param name="userInfo">A JSON object with all data necessary to create a new user account</param>
+        /// <returns>Returns a UserDTO for the newly created account. Throws an HTTP 409 if the
+        /// username or email is already in use.</returns>
+        [AllowAnonymous]
+        [HttpPost]
+        public async Task<ActionResult<UserDTO>> CreateNewUser([FromBody] NewUserDTO userInfo)
+        {
+            try
+            {
+                return await _userService.CreateNewUserAsync(userInfo);
+            }
+            catch(DbDetailedFailedWrite err)
+            {
+                return Conflict(new {message = err.Message}); // I would rather not tell which element is wrong (again someone might phish for valid emails), but maybe it is not a problem.
+            }
+        }
+        
+        /// <summary>
+        /// Confirms creation of new user account.
+        /// </summary>
+        /// <param name="payload">JSON object with token from user registration email.</param>
+        /// <returns>Returns 200 for success and 404 for failure.</returns>
```

Comment:
If the result is empty, we should return a 204.

Reply:
Thanks, I have updated the code and the description in our Website API Google Doc.